aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -425,19 +426,22 @@ public void observe(
     @Override
     public void clear(@NonNull Action onComplete,
                       @NonNull Consumer<DataStoreException> onError) {
-        beforeOperation(() -> {
-            orchestrator.stop();
-            sqliteStorageAdapter.clear(onComplete, onError);
-        });
+        // We shouldn't call call beforeOperation when clearing the DataStore. The
+        // only thing we have to wait for is the category initialization latch.
+        try {
+            categoryInitializationsPending.await();
```

Comment:
Should we be checking the return value here to tell whether or not the latch did count down? What are the appropriate actions to take if so, vs. if not?

Reply:
Let me think about this one. Just realized also that we don't have a timeout on this await. We'll need to do some testing on this though.

Reply:
I added a timeout and it's also now checking the result of the await operation.